File: server/routes/station.py
from flask import Blueprint, jsonify

import validators
from services.gtfs_client import get_stop_by_code

station_bp = Blueprint("station", __name__)


@station_bp.route("/station/<station_code>")
def get_station(station_code: str):
    """
    GET /station/<station_code>

    { "id": "...", "code": "1111", "name": "שם התחנה", "lat": 32.0, "lon": 34.8 }
    """
    code = validators.station_code(station_code)

    stop = get_stop_by_code(code)
    if stop is None:
        return jsonify({"error": f"תחנה {code} לא נמצאה"}), 404
    return jsonify(stop)

File: mobile/types/station.ts
export interface Station {
  id: string;
  name: string;
  code: string;
  lat?: number;
  lon?: number;
  /** מטרים מנקודת החיפוש. קיים רק בתשובת /nearby. */
  distance?: number;
}

/** תחנה עם קואורדינטות מובטחות — לשימוש במפה. */
export type LocatedStation = Station & { lat: number; lon: number };

export function hasCoords(station: Station): station is LocatedStation {
  return typeof station.lat === 'number' && typeof station.lon === 'number';
}
